src/libCli/OutputFormatting.hpp

```diff
@@ -108,17 +119,25 @@ namespace cli
                 std::string stringFromUInt(T f_value, const CustomStringModifier & f_modifier)
                 {
                     std::string result;
-                    if(f_modifier == CustomStringModifier::Raw)
+                    switch(f_modifier)
                     {
+                    case CustomStringModifier::Hex:
                         result += intToHexString(f_value);
-                    }
-                    else
-                    {
+                        break;
+                    case CustomStringModifier::Bin:
+                        dumpBinaryIntoString(result, f_value); // TODO: on a little endian client machine, this will be dumped out as LE. This is not wrong,
+                        break;                                //        but if the host is BE it might not be the expected behavior. What do we choose?
```

Comment:
neither. an integer can easily be converted to binary by looking at MSB. if 1 append 1 to string otherwise 0. then shoft left by 1 and repeat until empty.
this works platform independently

Reply:
OK, I realize I have a different understanding of binary than you have.... I thought binary is the human readable integer representation's base.
I would stay consistent with Hex and Dec here i.e. bin of 3 will give the string "11"
We sould then add an other modifier "Raw" which outputs raw "binary" data. this should NOT be souuprted for integers at all, as nobody will want to use this on integers because of endianness confusion. this modifier should only work on Bytes proto fields.

Reply:
Renaming bin to raw to clear that up.
However, imo printing integers as raw is beneficial if the output is well defined. So I'd vote for big endian or introduce a modifier parameter that explicitly states the endianess.
For example: `/myInt%raw,bigEndian/`